lib/archethic_web/graphql_schema/resolver.ex

```diff
@@ -33,6 +34,64 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
     end
   end
 
+  def get_token(address) do
+    tasks = [
+      Task.async(fn -> TransactionChain.fetch_genesis_address_remotely(address) end),
+      Task.async(fn -> get_transaction_content(address) end)
+    ]
+
+    # Using 5000 ms default timeout on tasks
+    genesis_address =
+      case Task.yield(List.first(tasks)) do
+        {:ok, {:ok, genesis_address}} -> genesis_address
+        {:ok, {:error, :network_issue}} -> :network_issue
+        _ -> nil
+      end
+
+    transaction_content =
+      case Task.yield(List.last(tasks)) do
+        {:ok, {:ok, content}} -> content
+        _ -> nil
+      end
+
+    if transaction_content != nil and genesis_address != nil do
```

Comment:
I think you could handle it with the `with` statement.

You could do something like this
```elixir

t1 = Task.async(fn -> TransactionChain.fetch_genesis_address_remotely(address) end)
t2 = Task.async(fn -> get_transaction_content(address) end)

with {:ok, genesis_address} <- Task.yield(t1),
     {:ok, tx_content} <- Task.yield(t2) do
    {:ok,
       %{
         genesis: genesis_address,
         name: Map.get(transaction_content, "name"),
         supply: Map.get(transaction_content, "supply"),
         symbol: Map.get(transaction_content, "symbol"),
         type: Map.get(transaction_content, "type"),
         properties: do_reduce_properties(Map.get(transaction_content, "properties"))
       }}
else
  {:error, :network_issue} ->
     {:error, "Network issue"}
  {:error, :invalid_transaction} ->
     {:error, "Transaction is not a token"}
  {:error, :transaction_not_exists} ->
     {:error, "Transaction doesn't exists"}
end
```

Reply:
Just to improve the readability

Reply:
Yes that's much more elegant and better, Thanks. Made the changes